packages/devtools_app/lib/src/debugger/debugger_controller.dart

```diff
@@ -624,31 +627,27 @@ class DebuggerController extends DisposableController
 
     // Perform an early exit if we're not paused.
     if (!paused) {
-      _populateFrameInfo([]);
+      _populateFrameInfo([], truncated: false);
       return;
     }
 
-    // First, notify based on the single 'pauseEvent.topFrame' frame.
-    if (pauseEvent?.topFrame != null) {
-      final tempFrames = _framesForCallStack(
-        [pauseEvent.topFrame],
-        reportedException: pauseEvent?.exception,
-      );
-      _populateFrameInfo(
-        [await _createStackFrameWithLocation(tempFrames.first)],
-        truncated: true,
-      );
-      _log.log('created first frame');
-    }
+    // We populate the first 12 frames; this ~roughly corresponds to the number
+    // of visible stack frames.
+    const initialFrameRequestCount = 12;
```

Comment:
is 12 enough if devtools is opened on a large screen (4k monitor for example)?

Reply:
If we want to make this smarter you can add a LayoutBuilder to the debugger UI and send a hint to the controller with the # of frames that will fit in the area each time the layout changes.

Reply:
> is 12 enough if devtools is opened on a large screen (4k monitor for example)? what is the added cost to make this larger than 12?

It probably wouldn't be large enough; the user would see some blank space at the bottom of the call stack which would fill in after the rest of the frames were available - they'd see a flash of content filling in at the bottom.

We could arbitrarily make it larger - populating a few more frames would only slow things down proportionally. We had used 10 here before. Any fixed value would be somewhat arbitrary. 12? 20?

> If we want to make this smarter you can add a LayoutBuilder to the debugger UI and send a hint to the controller with the # of frames that will fit in the area each time the layout changes.

This sounds reasonable, though we'd have to measure the height of a row here. I'd like to avoid more complexity here however. I think something like https://github.com/dart-lang/sdk/issues/43827 would be a good end state. We'd be able to just request all the frames when paused w/o having to limit to the first n frames.

Reply:
Agreed it doesn't make sense to implement as dart-lang/sdk#43827  should be cheap to implement and a better solution.